homeassistant/components/neato/__init__.py

```diff
@@ -77,6 +78,21 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         )
     )
 
+    if implementation.client_id != entry.unique_id:
+        _LOGGER.debug("Adding unique_id to old config entry")
+        hass.config_entries.async_update_entry(
+            entry, unique_id=implementation.client_id
```

Comment:
Two things:

- New config entries don't get this unique ID in the config flow either, meaning this migration path will never become obsolete at this point 😉 
- With OAuth you could add multiple instances using the same client_id & secret set, at least, that is how it is meant to be used according to specifications. By using it as a unique ID here, that kinda makes that impossible (or erroneous?)

Reply:
> * New config entries don't get this unique ID in the config flow either, meaning this migration path will never become obsolete at this point 😉

Will fix it.

> * With OAuth you could add multiple instances using the same client_id & secret set, at least, that is how it is meant to be used according to specifications. By using it as a unique ID here, that kinda makes that impossible (or erroneous?)

I was under the impression that Neato associates a `client_id` to a `redirect_url` and thus to a HA instance. Is able to fetch all robots from this single API call, so I don't see a point to get multiple instances.

But would be happy to change that to something more wise: suggestions ? ;-)

Simone

Reply:
So, if you have 2 Neato accounts, how does that work?

Reply:
> So, if you have 2 Neato accounts, how does that work?

I read somewhere that `redirect_url` would be unique in their system, but I may be easily wrong.
Let's go for something different, so we don't limit usage for such edge cases.

Reply:
As per your suggestion, coded using user id.

Simone